# Veterinary Radiology Intelligence

## Notebook 01: Data Understanding

### Objectives

The purpose of this notebook is to understand the structure and quality of the veterinary radiology dataset before performing any data cleaning or statistical analysis.

The exploratory analysis aims to:

- Understand the available variables.
- Identify missing values.
- Detect inconsistent data.
- Evaluate data quality.
- Document potential cleaning actions.

In [1]:
# ==========================================
# IMPORT LIBRARIES
# ==========================================

import pandas as pd

In [2]:
# ==========================================
# PROJECT PATHS
# ==========================================

from pathlib import Path

PROJECT_DIR = Path.cwd().parent
DATA_DIR = PROJECT_DIR / "data"
RAW_DATA_DIR = DATA_DIR / "raw"

print("Project folder:", PROJECT_DIR)
print("Raw data folder:", RAW_DATA_DIR)

Project folder: c:\Users\angie\OneDrive\Desktop\Veterinary-Radiology-Intelligence
Raw data folder: c:\Users\angie\OneDrive\Desktop\Veterinary-Radiology-Intelligence\data\raw


In [3]:
Path.cwd()

WindowsPath('c:/Users/angie/OneDrive/Desktop/Veterinary-Radiology-Intelligence/notebooks')

In [4]:
# ==========================================
# LOAD THE DATASET
# ==========================================

file_name = "CUADERNO RX.xlsx"

file_path = RAW_DATA_DIR / file_name

radiology_df = pd.read_excel(file_path)

print("Dataset loaded successfully!")

Dataset loaded successfully!


c:\Users\angie\AppData\Local\Programs\Python\Python313\Lib\site-packages\openpyxl\worksheet\_reader.py:223: UserWarning: Cell B659 is marked as a date but the serial value 6692321.0 is outside the limits for dates. The cell will be treated as an error.
  warn(msg)


In [5]:
print("Dataset loaded successfully!")

Dataset loaded successfully!


In [7]:
# ==========================================
# DISPLAY THE FIRST ROWS
# ==========================================

radiology_df.head()

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
0,4315,2022-01-18 00:00:00,LIMA,FELINO,CRIOLLO,2021-09-02 00:00:00,3,H,NS,CAMILO RAMIREZ,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,4316,2021-02-19 00:00:00,THORE,CANINO,BULL DOG INGLES,2020-10-01 00:00:00,16,M,SI,CAROLINA SUAREZ GARZON,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,4317,2021-02-19 00:00:00,ESTRELLITA,CANINO,BEAGLE,2013-02-01 00:00:00,11,H,SI,LORENA RODRIGUEZ,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,4318,2022-02-19 00:00:00,MILU,CANINO,PASTOR ALEMAN,01.01.2013,23,H,NO,DUVAN PEÑA,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,4319,2022-02-21 00:00:00,TAYLER,CANINO,PIT BULL,01.01.2020,28,M,NO,LEONARDO CUELLAR,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [8]:
# ==========================================
# DATASET DIMENSIONS
# ==========================================

print(f"Rows: {radiology_df.shape[0]}")
print(f"Columns: {radiology_df.shape[1]}")

Rows: 864
Columns: 25


In [10]:
radiology_df.head()


,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
0,4315,2022-01-18 00:00:00,LIMA,FELINO,CRIOLLO,2021-09-02 00:00:00,3,H,NS,CAMILO RAMIREZ,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,4316,2021-02-19 00:00:00,THORE,CANINO,BULL DOG INGLES,2020-10-01 00:00:00,16,M,SI,CAROLINA SUAREZ GARZON,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,4317,2021-02-19 00:00:00,ESTRELLITA,CANINO,BEAGLE,2013-02-01 00:00:00,11,H,SI,LORENA RODRIGUEZ,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,4318,2022-02-19 00:00:00,MILU,CANINO,PASTOR ALEMAN,01.01.2013,23,H,NO,DUVAN PEÑA,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,4319,2022-02-21 00:00:00,TAYLER,CANINO,PIT BULL,01.01.2020,28,M,NO,LEONARDO CUELLAR,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [11]:
radiology_df.shape

(864, 25)

In [12]:
radiology_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 25 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   21                  864 non-null    int64  
 1   FECHA               692 non-null    object 
 2   NOMBRE MASCOTA      695 non-null    object 
 3   ESPECIE             694 non-null    object 
 4   RAZA                693 non-null    object 
 5   FECHA  NACIMIENTO   691 non-null    object 
 6   PESO                666 non-null    object 
 7   SEXO                692 non-null    object 
 8   ¿ESTERIL?           683 non-null    object 
 9   NOMBRE PROPIETARIO  693 non-null    object 
 10  DR. REMITENTE       690 non-null    object 
 11  CANTIDAD RX         659 non-null    object 
 12  ¿REPORTE?           664 non-null    object 
 13  CONTACTO            683 non-null    object 
 14  Unnamed: 14         0 non-null      float64
 15  Unnamed: 15         0 non-null      float64
 16  Unnamed:

## Initial Findings

### General information

- The dataset contains **864 records** and **25 columns**.

### Data quality observations

- Several columns (`Unnamed:14` to `Unnamed:23`) are completely empty.
- The first column is named `21`, which requires further investigation.
- Several columns contain missing values.
- Date columns are currently stored as `object`.
- Weight is also stored as `object`.

In [13]:
# ==========================================
# DISPLAY COLUMN NAMES
# ==========================================

for column in radiology_df.columns:
    print(column)

21
FECHA
NOMBRE MASCOTA
ESPECIE
RAZA
FECHA  NACIMIENTO
PESO
SEXO
¿ESTERIL?
NOMBRE PROPIETARIO
DR. REMITENTE
CANTIDAD RX
¿REPORTE?
CONTACTO
Unnamed: 14
Unnamed: 15
Unnamed: 16
Unnamed: 17
Unnamed: 18
Unnamed: 19
Unnamed: 20
Unnamed: 21
Unnamed: 22
Unnamed: 23
Unnamed: 24


## Next Steps

The next step is to inspect each column individually to understand:

- Data consistency
- Missing values
- Unique values
- Possible typing errors
- Data quality issues

In [15]:
# ==========================================
# DATASET SUMMARY
# ==========================================

summary_df = pd.DataFrame({
    "Data Type": radiology_df.dtypes,
    "Missing Values": radiology_df.isna().sum(),
    "Non-Null Values": radiology_df.notna().sum(),
    "Unique Values": radiology_df.nunique()
})
summary_df

,Data Type,Missing Values,Non-Null Values,Unique Values
21,int64,0,864,864
FECHA,object,172,692,247
NOMBRE MASCOTA,object,169,695,482
ESPECIE,object,170,694,18
RAZA,object,171,693,110
FECHA NACIMIENTO,object,173,691,383
PESO,object,198,666,182
SEXO,object,172,692,7
¿ESTERIL?,object,181,683,9
NOMBRE PROPIETARIO,object,171,693,676


In [16]:
# ==========================================
# CHECK DUPLICATE RECORDS
# ==========================================

duplicate_rows = radiology_df.duplicated().sum()

print(f"Duplicate rows: {duplicate_rows}")

Duplicate rows: 0


In [17]:
radiology_df.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
859    False
860    False
861    False
862    False
863    False
Length: 864, dtype: bool

In [19]:
# ==========================================
# CHECK DUPLICATE RECORDS
# ==========================================

duplicate_rows = radiology_df.duplicated().sum()

print(f"Duplicate rows: {duplicate_rows}")

Duplicate rows: 0


### Duplicate Records Assessment

No fully duplicated records were found in the dataset.

This indicates that each record is unique when considering all available columns.

Further analysis will focus on identifying duplicated patients or repeated visits, which are different from duplicated records.

In [20]:
radiology_df.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
859    False
860    False
861    False
862    False
863    False
Length: 864, dtype: bool

In [21]:
# ==========================================
# SPECIES FREQUENCY
# ==========================================

species_counts = radiology_df["ESPECIE"].value_counts(dropna=False)

species_counts

ESPECIE
CANINO      474
FELINO      181
NaN         170
CANINO       15
FELINO        7
FELINNO       2
HUMANO        2
CANINNO       2
FELIINO       1
              1
FELINNNO      1
CSNINO        1
ÑCANNO        1
CANINo        1
 CANINO       1
CANELA        1
FELINO\n      1
CONEJO        1
CAINO         1
Name: count, dtype: int64

In [22]:
radiology_df["ESPECIE"]

0      FELINO
1      CANINO
2      CANINO
3      CANINO
4      CANINO
        ...  
859       NaN
860       NaN
861       NaN
862       NaN
863       NaN
Name: ESPECIE, Length: 864, dtype: object

In [24]:
radiology_df["ESPECIE"].value_counts()

ESPECIE
CANINO      474
FELINO      181
CANINO       15
FELINO        7
HUMANO        2
FELINNO       2
CANINNO       2
              1
FELINNNO      1
FELIINO       1
ÑCANNO        1
CANINo        1
CSNINO        1
 CANINO       1
CANELA        1
FELINO\n      1
CONEJO        1
CAINO         1
Name: count, dtype: int64

## Species Assessment

### Findings

- The `ESPECIE` column contains spelling inconsistencies.
- Several values represent the same species but are written differently (e.g., `CANINO`, `CANINNO`, `CAINO`, `CANINo`).
- Similar inconsistencies were found for `FELINO`.
- Some unexpected values (`HUMANO`, `CANELA`) require further investigation.
- The column also contains missing values and blank entries.
- A standardization process will be required during the data cleaning phase.

In [25]:
# ==========================================
# INVESTIGATE SPECIES VALUES
# ==========================================

radiology_df[radiology_df["ESPECIE"] == "HUMANO"]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
366,4700,2022-08-18 00:00:00,SOFIA,HUMANO,HUMANO,NS,NS,H,NO,MAYRA GARCIA,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
636,4970,2022-12-05 00:00:00,ALEJANDRO PUERTA,HUMANO,HUMANO,06.08.1987,NaN,M,NO,ALEJANDRO PUERTA,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


### Human Records

Two records were found with the species classified as `HUMANO`.

At this stage, it is not possible to determine whether these records correspond to data entry errors, equipment testing, or intentional human radiographic studies.

These records will be preserved until additional domain knowledge is available.

In [26]:
# ==========================================
# FIND UNEXPECTED SPECIES
# ==========================================

expected_species = ["CANINO", "FELINO", "CONEJO"]

unexpected_species = radiology_df[
    ~radiology_df["ESPECIE"].isin(expected_species)
]

unexpected_species[["ESPECIE", "NOMBRE MASCOTA", "RAZA"]]

,ESPECIE,NOMBRE MASCOTA,RAZA
45,FELIINO,TOMAS,CRIOLLO
201,FELINO,LIMON,MESTIZO
263,CANINO,MAX,BULL TERRIER
290,FELINNNO,RAMON,CRIOLLO
321,,PRINCESA,PINCHER
...,...,...,...
859,NaN,NaN,NaN
860,NaN,NaN,NaN
861,NaN,NaN,NaN
862,NaN,NaN,NaN


In [27]:
~radiology_df["ESPECIE"].isin(expected_species)

0      False
1      False
2      False
3      False
4      False
       ...  
859     True
860     True
861     True
862     True
863     True
Name: ESPECIE, Length: 864, dtype: bool

In [28]:
expected_species = ["CANINO", "FELINO", "CONEJO"]

In [29]:
# ==========================================
# FIND UNEXPECTED SPECIES
# ==========================================

expected_species = ["CANINO", "FELINO", "CONEJO"]

unexpected_species = radiology_df[
    ~radiology_df["ESPECIE"].isin(expected_species)
]

unexpected_species[["ESPECIE", "NOMBRE MASCOTA", "RAZA"]]

,ESPECIE,NOMBRE MASCOTA,RAZA
45,FELIINO,TOMAS,CRIOLLO
201,FELINO,LIMON,MESTIZO
263,CANINO,MAX,BULL TERRIER
290,FELINNNO,RAMON,CRIOLLO
321,,PRINCESA,PINCHER
...,...,...,...
859,NaN,NaN,NaN
860,NaN,NaN,NaN
861,NaN,NaN,NaN
862,NaN,NaN,NaN


In [30]:
# ==========================================
# UNIQUE SPECIES VALUES
# ==========================================

species_list = sorted(radiology_df["ESPECIE"].dropna().unique())

for species in species_list:
    print(repr(species))

' '
' CANINO'
'CAINO'
'CANELA'
'CANINNO'
'CANINO'
'CANINO '
'CANINo'
'CONEJO'
'CSNINO'
'FELIINO'
'FELINNNO'
'FELINNO'
'FELINO'
'FELINO\n'
'FELINO '
'HUMANO'
'ÑCANNO'


### Conclusion

The `ESPECIE` column presents several data quality issues:

- Typographical errors
- Leading/trailing spaces
- Different capitalization
- Missing values
- Unexpected categories

This column will require standardization during the data cleaning phase.

## Species Assessment

### Data Quality Issues

The `ESPECIE` column presents several data quality problems:

- Missing values.
- Blank spaces used as values.
- Leading and trailing spaces.
- Inconsistent capitalization.
- Typographical errors.
- Unexpected categories.
- Hidden characters (newline characters).

These issues will be addressed during the data cleaning phase.

In [31]:
# ==========================================
# SEX DISTRIBUTION
# ==========================================

sex_counts = radiology_df["SEXO"].value_counts(dropna=False)

sex_counts

SEXO
M      370
H      309
NaN    172
NS       5
M        4
H        2
SI       1
NH       1
Name: count, dtype: int64

## Sex Assessment

### Data Quality Issues

The `SEXO` column is generally consistent, with `M` and `H` representing the majority of the records.

However, the following issues were identified:

- Missing values.
- Possible leading or trailing spaces.
- Unexpected values (`NS`, `SI`, `NH`).
- The meaning of some uncommon values requires further investigation.

The column will be standardized during the data cleaning phase.

In [32]:
# ==========================================
# STERILIZATION STATUS DISTRIBUTION
# ==========================================

sterilization_counts = radiology_df["¿ESTERIL?"].value_counts(dropna=False)

sterilization_counts

¿ESTERIL?
SI      356
NO      230
NaN     181
NS       83
NO        4
SI        4
Si        3
SNO       1
NO R      1
Na        1
Name: count, dtype: int64

In [ ]:
## Sterilization Status Assessment

### Data Quality Issues

The `¿ESTERIL?` column is generally consistent, with `SI` and `NO` representing the majority of the records.

However, several data quality issues were identified:

- Missing values.
- Leading and trailing spaces.
- Inconsistent capitalization (`SI` vs `Si`).
- Unexpected values (`NS`, `SNO`, `NO R`, `Na`).

These values will be investigated and standardized during the data cleaning phase.

In [33]:
# ==========================================
# WEIGHT FREQUENCY
# ==========================================

weight_counts = radiology_df["PESO"].value_counts(dropna=False)

weight_counts.head(30)

PESO
NaN    198
NS      67
3       41
4       38
6       28
12      19
8       18
2       14
20      12
18      12
5       12
28      11
32      10
25       9
16       9
30       9
10       9
11       9
13       8
7        7
9        7
15       7
4.2      7
7.4      6
4.4      6
3.8      6
6.2      6
17       5
14       5
3.2      5
Name: count, dtype: int64

In [35]:
# ==========================================
# DATA TYPES INSIDE THE WEIGHT COLUMN
# ==========================================

radiology_df["PESO"].apply(type).value_counts()

PESO
<class 'float'>    426
<class 'int'>      361
<class 'str'>       77
Name: count, dtype: int64

In [36]:
# ==========================================
# DISPLAY TEXT VALUES IN WEIGHT
# ==========================================

text_weights = radiology_df[
    radiology_df["PESO"].apply(lambda x: isinstance(x, str))
]

text_weights["PESO"].value_counts()

PESO
NS       67
ns        1
26kg      1
11,80     1
3,2       1
SI        1
4,4       1
6.20.     1
NO        1
8,6       1
11,60     1
Name: count, dtype: int64

## Weight Assessment

### Domain Knowledge

Based on veterinary practice, some values have a specific clinical meaning:

- `NS` indicates that the patient's weight was **not recorded**, because it was not required during the examination.
- `NO` indicates that **no weight was determined** at the time of the examination.
- Decimal values written with commas follow the local numeric notation and represent valid weights.
- Values such as `26kg` include the measurement unit and require standardization.

These observations will guide the data cleaning process in the next notebook.

## Domain Knowledge

This project was developed using real veterinary clinical records.

The data cleaning strategy was not based only on programming techniques, but also on veterinary clinical knowledge to correctly interpret missing values, abbreviations, and data entry practices.

This approach improves data quality while preserving the clinical meaning of the records.

In [38]:
# ==========================================
# DATE DATA TYPES
# ==========================================

radiology_df["FECHA"].apply(type).value_counts()

FECHA
<class 'datetime.datetime'>    674
<class 'float'>                172
<class 'str'>                   18
Name: count, dtype: int64

In [39]:
# ==========================================
# MISSING DATES
# ==========================================

missing_dates = radiology_df["FECHA"].isna().sum()

print(f"Missing dates: {missing_dates}")

Missing dates: 172


In [40]:
# ==========================================
# SAMPLE DATE VALUES
# ==========================================

radiology_df["FECHA"].dropna().head(20)

0     2022-01-18 00:00:00
1     2021-02-19 00:00:00
2     2021-02-19 00:00:00
3     2022-02-19 00:00:00
4     2022-02-21 00:00:00
5     2022-02-21 00:00:00
6     2022-02-22 00:00:00
7     2022-02-24 00:00:00
8     2022-02-25 00:00:00
9     2022-02-26 00:00:00
10    2022-02-26 00:00:00
11    2022-02-27 00:00:00
12    2022-02-27 00:00:00
13    2022-02-27 00:00:00
14    2022-02-27 00:00:00
15    2022-02-28 00:00:00
16    2021-02-28 00:00:00
17    2022-02-28 00:00:00
18    2022-03-01 00:00:00
19    2022-03-02 00:00:00
Name: FECHA, dtype: object

In [41]:
# ==========================================
# DISPLAY TEXT DATES
# ==========================================

text_dates = radiology_df[
    radiology_df["FECHA"].apply(lambda x: isinstance(x, str))
]

text_dates["FECHA"]

135    25/040/2022
139     26.04.2022
141     27.04.2022
142     27.04.2022
143     29/0472022
167     14/0572022
506     12.10.2022
507     12.10.2022
508     13.10.2022
509     13.10.2022
510     13.10.2022
511     13.10.2022
519      19/102022
524     20.10.2022
642      8.12.2022
655     13.12.2022
656     13.12.2022
668    21/12//2022
Name: FECHA, dtype: object

In [42]:
# ==========================================
# UNIQUE TEXT DATES
# ==========================================

text_dates["FECHA"].value_counts()

FECHA
13.10.2022     4
27.04.2022     2
12.10.2022     2
13.12.2022     2
29/0472022     1
26.04.2022     1
25/040/2022    1
14/0572022     1
19/102022      1
20.10.2022     1
8.12.2022      1
21/12//2022    1
Name: count, dtype: int64

## Date Assessment

### Data Quality Assessment

The `FECHA` column shows a high level of data quality.

Most records were automatically recognized as valid datetime values by Pandas.

Only a small number of records remain as text due to formatting inconsistencies.

The following issues were identified:

- Dates written using periods (`13.10.2022`).
- Missing separators (`29/0472022`).
- Extra separators (`21/12//2022`).
- Minor typing errors (`25/040/2022`).

Based on the observed patterns, these records appear to be data entry errors rather than missing or invalid dates.

The affected records will be standardized during the data cleaning phase

In [43]:
# ==========================================
# REFERRING VETERINARIAN FREQUENCY
# ==========================================

veterinarian_counts = radiology_df["DR. REMITENTE"].value_counts(dropna=False)

veterinarian_counts.head(30)

DR. REMITENTE
NaN                     174
ERWIN RESTREPO           78
RUBEN GOMEZ              35
ANIBAL GARCIA            33
JAIME VEGA               22
ANDRES RUIZ              22
JAIME VEGA               18
JUAN CAMILO TRUJILLO     17
ERWIN RESTREPO           14
JUAN PABLO ORTIZ         13
LUISA VARGAS             13
JULIO CESAR DUARTE       13
DR ANIBAL GARCIA         13
DR ERWIN RESTREPO        13
DR JAIME VEGA            10
DR JUAN CAMILO           10
MATEOS                   10
VET MATEOS               10
MARIBEL MUÑOZ             9
EDGAR ORDOÑEZ             8
ALEJANDRO PUERTA          8
SANDRA TORRES             7
DRA MAYRA GARCIA          7
JUAN CAMILO               6
RAFAEL TORRES             6
ANIBAL GARCIA             6
DR ERWIN                  6
ALEJANDRO PUERTA          6
MAIRA GARCIA              5
DR RUBEN GOMEZ            5
Name: count, dtype: int64

In [44]:
# ==========================================
# NUMBER OF UNIQUE VETERINARIANS
# ==========================================

radiology_df["DR. REMITENTE"].nunique()

195

In [45]:
# ==========================================
# UNIQUE VETERINARIAN NAMES
# ==========================================

veterinarians = sorted(
    radiology_df["DR. REMITENTE"].dropna().astype(str).unique()
)

for veterinarian in veterinarians:
    print(repr(veterinarian))

'ADRIAN AOSORNO'
'ADRIAN MARCELA OSORNO'
'ADRIANA MARCELA OSORNO'
'ADRIANA OSORNO'
'ADRIANA OSORNO '
'ADRIANA OSRONO'
'ALEJANDRO PUERTA'
'ALEJANDRO PUERTA '
'ALEJANNDRO PUERTA'
'AMBULANCIA ANIMAL'
'AMBULANCIA ANIMAL '
'AMIGOS DE 4 PATAS'
'ANDRES GARCÍA '
'ANDRES RIUZ'
'ANDRES RIZ Y MARIA BAUTISTA'
'ANDRES RUIZ'
'ANDRES RUIZ '
'ANDRES RUIZ MARIA BAUTISTA'
'ANDRES RUIZ Y MARIA'
'ANDRES RUIZ Y MARIA BAUTISTA'
'ANDRES SIERRA'
'ANDRES Y MARIA'
'ANIBAL GARCIA'
'ANIBAL GARCIA '
'ANIMALS HELP'
'CARLOS ALBERTO GARCIA'
'CARLOS HERNANDEZ'
'CARLOS HERNANDEZ LEON'
'CESAR TORRES'
'CHANI COCKER '
'CHANICOCKER'
'CHANICOCKER '
'CUATRO PATAS'
'DANIEL MENDIETA'
'DR ALEJANDRO PUERTA'
'DR ANBIBAL GARCIA'
'DR ANDRES RUIZ'
'DR ANIBAL'
'DR ANIBAL GARCIA'
'DR CARLOS SILVA'
'DR DANIEL MARTINEZ'
'DR DANIEL MENDIETA'
'DR EDGAR ORDOÑEZ'
'DR ERVIN'
'DR ERWIN'
'DR ERWIN RESTREPO'
'DR ERWIN RESTREPO '
'DR JAIME VEGA'
'DR JUA CAMILO TRUJILLO'
'DR JUAN CAMILO'
'DR JUAN CAMILO TRUJILLO'
'DR JUAN PABLO'
'DR JULIO CESAR'


## Referring Veterinarian Assessment

### Data Quality Assessment

The `DR. REMITENTE` column presents significant data quality issues.

The same referring veterinarian appears under multiple name variations.

The identified inconsistencies include:

- Leading and trailing spaces.
- Different prefixes (`DR`, `DRA`, `VET`).
- Spelling mistakes.
- Partial names.
- Different representations of the same veterinary clinic.

These inconsistencies generate duplicate entities and would lead to inaccurate statistics if left uncorrected.

The column will require standardization before database creation and dashboard development.

In [46]:
# ==========================================
# RADIOGRAPH QUANTITY FREQUENCY
# ==========================================

rx_quantity_counts = radiology_df["CANTIDAD RX"].value_counts(dropna=False)

rx_quantity_counts

CANTIDAD RX
2      363
1      263
NaN    205
3       19
4        7
NS       3
5        2
6        1
0        1
Name: count, dtype: int64

In [47]:
# ==========================================
# RADIOGRAPH QUANTITY DATA TYPES
# ==========================================

radiology_df["CANTIDAD RX"].apply(type).value_counts()

CANTIDAD RX
<class 'int'>      656
<class 'float'>    205
<class 'str'>        3
Name: count, dtype: int64

In [48]:
# ==========================================
# UNIQUE RADIOGRAPH QUANTITY VALUES
# ==========================================

for value in radiology_df["CANTIDAD RX"].dropna().unique():
    print(repr(value))

5
2
6
1
3
4
'NS'
0


## Radiograph Quantity Assessment

### Data Quality Assessment

The `CANTIDAD RX` column shows a high level of data quality.

Most records are stored as integer values representing the number of radiographs performed.

The identified issues include:

- Missing values.
- Three records labeled as `NS`.
- One record with a value of `0`, which requires further investigation.

Overall, the column is suitable for quantitative analysis after minor data cleaning.

In [49]:
# ==========================================
# INVESTIGATE ZERO RADIOGRAPHS
# ==========================================

radiology_df[radiology_df["CANTIDAD RX"] == 0]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
392,4726,2022-08-29 00:00:00,PAUL,CANINO,SHIH TZU,02.02.2022,4.2,M,NO,MILLEN ESTEBAN MUYANO,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


### Observation

One record contains a radiograph quantity equal to `0`.

The record is otherwise complete and does not appear to be incomplete.

Since there is no evidence that the value is incorrect, it will be preserved during the exploration phase and reviewed during data cleaning if additional information becomes available.

In [50]:
# ==========================================
# RADIOLOGY REPORT DISTRIBUTION
# ==========================================

report_counts = radiology_df["¿REPORTE?"].value_counts(dropna=False)

report_counts

¿REPORTE?
NO            347
SI            297
NaN           200
NO              5
NS              5
revisar         4
SI              3
SI (04443)      1
1               1
TORAX           1
Name: count, dtype: int64

In [51]:
# ==========================================
# REPORT COLUMN DATA TYPES
# ==========================================

radiology_df["¿REPORTE?"].apply(type).value_counts()

¿REPORTE?
<class 'str'>      663
<class 'float'>    200
<class 'int'>        1
Name: count, dtype: int64

In [52]:
# ==========================================
# UNIQUE REPORT VALUES
# ==========================================

for value in radiology_df["¿REPORTE?"].dropna().unique():
    print(repr(value))

'SI'
'NO'
'NO '
'NS'
'SI (04443)'
'TORAX'
1
'SI '
'revisar'


In [53]:
# ==========================================
# INVESTIGATE "REVISAR"
# ==========================================

radiology_df[radiology_df["¿REPORTE?"] == "revisar"]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
647,4981,2022-12-09 00:00:00,VENUS,CONEJO,CRIOLLO,09.12.2015,3,M,SI,DIANA CABEZAS,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
649,4983,2022-12-09 00:00:00,JONAS,FELINO,CRIOLLO,7.06.2012,4,H,SI,JAZMIN CASTRO,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
651,4985,2022-12-10 00:00:00,RUFITO,CANINO,FRENCH,8 AÑOS,9,M,SI,NELLY REYES,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
652,4986,2022-12-10 00:00:00,GINGER,FELINO,CRIOLLO,01.08.2022,2.2,H,NO,EDNA MUÑOZ,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [54]:
# ==========================================
# INVESTIGATE "SI (04443)"
# ==========================================

radiology_df[radiology_df["¿REPORTE?"] == "SI (04443)"]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
168,4488,2022-05-14 00:00:00,GOLDBERG,CANINO,COCKER,01.01.2014,12,M,NS,ROSMARY CABALLERO,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [55]:
# ==========================================
# INVESTIGATE "TORAX"
# ==========================================

radiology_df[radiology_df["¿REPORTE?"] == "TORAX"]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
328,4661,2022-08-01 00:00:00,KIRA,CANINO,BOSTON TERRIER,11 AÑOS,6,H,SI,ANDRES CORDOBA,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [56]:
# ==========================================
# INVESTIGATE VALUE 1
# ==========================================

radiology_df[radiology_df["¿REPORTE?"] == 1]

,21,FECHA,NOMBRE MASCOTA,ESPECIE,RAZA,FECHA NACIMIENTO,PESO,SEXO,¿ESTERIL?,NOMBRE PROPIETARIO,...,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18,Unnamed: 19,Unnamed: 20,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24
331,4664,2022-08-01 00:00:00,SISU,CANINO,SHIT ZU,1 AÑO,4,H,NO,LAURA CASTELLANOS,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


## Report Column Assessment

The `¿REPORTE?` column presents good overall data quality.

Most values correspond to the expected responses (`SI` and `NO`).

However, several isolated inconsistencies were identified:

- Leading/trailing spaces.
- Missing values.
- `NS` values indicating missing registration.
- Internal notes (`revisar`).
- One annotated value (`SI (04443)`).
- One numeric value (`1`).
- One anatomical term (`TORAX`) that appears to be misplaced.

These cases represent isolated data entry inconsistencies and will be reviewed during the data cleaning phase.

In [57]:
# ==========================================
# CONTACT INFORMATION DISTRIBUTION
# ==========================================

contact_counts = radiology_df["CONTACTO"].value_counts(dropna=False)

contact_counts

CONTACTO
NO                               630
NaN                              181
SI                                17
NO                                 4
NNO                                2
3023772027                         1
3112101288                         1
3168743939                         1
3123093900                         1
3112248627                         1
3156085941                         1
3203554721                         1
3232192285                         1
320 8306782                        1
3219205652                         1
3112687377                         1
3156937735                         1
3103396906                         1
3175022671                         1
3195928576                         1
3156949255                         1
3057487595                         1
3183562032                         1
3204458384 Yolanda                 1
3196248160                         1
3003775545                         1
3125943897                   

In [58]:
# ==========================================
# CONTACT COLUMN DATA TYPES
# ==========================================

radiology_df["CONTACTO"].apply(type).value_counts()

CONTACTO
<class 'str'>      658
<class 'float'>    181
<class 'int'>       25
Name: count, dtype: int64

In [59]:
# ==========================================
# UNIQUE CONTACT VALUES
# ==========================================

for value in radiology_df["CONTACTO"].dropna().unique():
    print(repr(value))

'NO'
3112101288
3023772027
3123093900
3156085941
3112248627
3168743939
3156937735
3103396906
3175022671
3203554721
3232192285
'320 8306782'
3219205652
3112687377
3057487595
3156949255
3195928576
3183562032
3196248160
3003775545
3125943897
'3204458384 Yolanda'
3112260390
3133346229
'direccion@sangregoriocota.org'
3185848093
'SI'
3203011998
3112208619
'NNO'
'NO '
'O'
'no'


## Contact Column Assessment

The `CONTACTO` column contains mixed information.

Most records use categorical values (`SI` and `NO`), while a smaller number contain phone numbers or additional notes.

This suggests inconsistent data entry practices over time.

The column will require standardization during the data cleaning phase.

# Final Assessment

## Dataset Summary

The veterinary radiology dataset contains 864 records and 25 columns.

The exploratory data analysis revealed that the dataset contains valuable clinical information but also several data quality issues that must be addressed before performing statistical analyses or building dashboards.

## Main Findings

- Several completely empty columns were identified.
- Missing values are present in multiple variables.
- Text fields contain typographical errors and inconsistent capitalization.
- Date formats are inconsistent.
- Some categorical variables include unexpected values.
- Veterinarian names are not standardized.
- The contact field mixes categorical values and phone numbers.
- Domain knowledge was essential to correctly interpret variables such as radiology reports and contact status.

## Overall Data Quality

Despite the inconsistencies, the dataset has high analytical value.

Most issues are related to manual data entry and can be addressed through systematic data cleaning.

## Next Steps

The next notebook will focus on data cleaning and standardization.

The cleaned dataset will later be stored in a SQL database and used to develop an interactive veterinary radiology dashboard.